python/cudf_polars/cudf_polars/dsl/expressions/boolean.py

```diff
@@ -170,7 +172,9 @@ def do_evaluate(
             is_finite = self.name is BooleanFunction.Name.IsFinite
             if child.dtype.id() not in (plc.TypeId.FLOAT32, plc.TypeId.FLOAT64):
                 value = plc.Scalar.from_py(is_finite)
-                return Column(plc.Column.from_scalar(value, df.num_rows))
+                return Column(
+                    plc.Column.from_scalar(value, df.num_rows), dtype=self.dtype
```

Comment:
Is `self.dtype` correct here (inside `IsFine / IsInfinite` of `do_evaluate`), or should it be polars' bool dtype?

Same question for the rest of the branches here.

Reply:
Makes sense, thanks.

Reply:
Yes, that is correct.